sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -194,6 +196,55 @@ export class StudentService extends RecordDataModelService<Student> {
       throw error;
     }
 
+    // If sfasIndividual wasn't found, check for a partial match
+    if (!sfasIndividual) {
```

Comment:
The logic to generate the notification should be extracted to a private method. It would also probably resolve the Sobarcloud issue about the [method's cognitive complexity](https://sonarcloud.io/project/issues?resolved=false&pullRequest=3218&id=bcgov_SIMS&open=AY8_1Z7JNln50TcTfFHy).

Reply:
updated